cmake/finders/FindCarlaUtils.cmake

```diff
@@ -0,0 +1,112 @@
+#[=======================================================================[.rst
+FindCarlaUtils
+----------
+
+FindModule for carla-utils and associated libraries
+
+Result Variables
+^^^^^^^^^^^^^^^^
+
+This module sets the following variables:
+
+``CARLAUTILS_FOUND``
```

Comment:
The finder is named `FindCarlaUtils`, thus all variables need to use the same spelling, i.e. `CarlaUtils_FOUND`. This applies to all variables throughout the file.